File: leetcode/Arrays_Hash/3Sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = set()
        nums.sort()

        for i in range(len(nums)):
            j = i + 1
            k = len(nums) - 1
            while (j < k):
                summ = nums[i] + nums[j] + nums[k]
                if (summ == 0):
                    res.add((nums[i], nums[j], nums[k])) 
                    j += 1
                    k -= 1
                elif (summ > 0):
                    k -= 1
                else:
                    j += 1
        res = list(res)
        return res      

File: leetcode/Arrays_Hash/3Sum.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        nums = nums.sort(nums.begin(), nums(end));
        set<vector<int>> res;
        sort(nums.begin(), nums.end());

        for (int i = 0; i < nums.size(); i++) {
            int j = i + 1;
            int k = nums.size() - 1;

            while (j < k) {
                int summ = nums[i] + nums[j] + nums[k];

                if (summ == 0) {
                    res.insert({nums[i], nums[j], nums[k]});
                    j++;
                    k--;
                }
                else if (summ > 0) {
                    k--;
                }
                else {
                    j++;
                }
            }
        }

        vector<vector<int>> result(res.begin(), res.end());
        return result;
    }
};